### ISSUE TITLE
Avoid using StopIteration inside of generator

### ISSUE THREAD

**Issue by polishmatt** [recurly/recurly-client-python#263]
Current behavior of StopIteration inside of a generator is accidental and has been deprecated, at least as of python 3.6:
https://docs.python.org/3.6/whatsnew/3.5.html#whatsnew-pep-479

Using StopIteration this way (at least in python 3.6) results in the following warning, and will result in a RuntimeError in future versions of python:
```
DeprecationWarning: generator 'Page.__iter__' raised StopIteration
```

Instead of raising a StopIteration, the Page class should use the return or yield keyword as appropriate:
https://github.com/recurly/recurly-client-python/blob/967a92a2f866174d204468f04b2865b311caa9fe/recurly/resource.py#L90

**Comment by polishmatt** [comment]
Specific examples of how to resolve the issue:
https://www.python.org/dev/peps/pep-0479/#consequences-for-existing-code

**Comment by aaron-suarez** [comment]
@polishmatt Thank you for bringing this up! We'll take a look at it and see if we can't refactor that. If you feel like you already know the solution for this particular generator, please feel free to open a PR and I'll be happy to review and merge it!

**Comment by aaron-suarez** [comment]
Fixed by #264

**Comment by bhelx** [comment]
@polishmatt thanks for contributing!
